From 631481d98bbc536cc64fbb15ac9f47d15c63037e Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 27 Jan 2022 14:13:14 +0200
Subject: [PATCH] [SW-0] preboot: add sizes.h from linux

Add sizes.h header with definitions of various sizes and update the
whole tree for using it.
This is to improve the readability of the magic numbers we have all
over.
Also extend the sizes.h with the values used in HBM code that are not
present in the base file.

Change-Id: I658ca5d7441b674049792464125796a32cee484d
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/arm/common.h b/arch/gaudi/arm/common.h
index e8401e2..60afdc2 100644
--- a/arch/gaudi/arm/common.h
+++ b/arch/gaudi/arm/common.h
@@ -8,6 +8,7 @@
 #define COMMON_GAUDI_H_
 
 #include <address_space.h>
+#include <sizes.h>
 
 #define PPBOOT_SIZE			0x5000		/* 20KB */
 #define PREBOOT_SIZE			0x25000		/* 148KB */
@@ -20,7 +21,7 @@
 #define PPBOOT_TEXT_BASE		0x7FF0000000
 #define PREBOOT_TEXT_BASE		(PPBOOT_TEXT_BASE + PPBOOT_SIZE)
 
-#define HL_SPI_SIZE			0x2000000
+#define HL_SPI_SIZE			SZ_32M
 #define HL_CFG_BASE			0x7FFC000000ull
 #define SDRAM_BASE			0x7FF0000000ull
 
diff --git a/arch/gaudi/arm/hbm.c b/arch/gaudi/arm/hbm.c
index e50e351..2e6e07a 100644
--- a/arch/gaudi/arm/hbm.c
+++ b/arch/gaudi/arm/hbm.c
@@ -21,6 +21,7 @@
 #include <pll.h>
 #include <uart.h>
 #include <string.h>
+#include <sizes.h>
 
 #include "common.h"
 
@@ -47,9 +48,6 @@
 #define EXTEST_READ_INST		0
 #define EXTEST_WDR_BYTE_OFFSET		12
 #define EXTEST_WDR_DWORD_OFFSET		48
-
-#define SZ_16G				_AC(0x400000000, ULL)
-#define SZ_32G				_AC(0x800000000, ULL)
 
 #define BYTES_PER_DWORD			4
 #define DWORDS_PER_PC			2
diff --git a/arch/gaudi/arm/rma_info.c b/arch/gaudi/arm/rma_info.c
index 0bf3bb6..110e748 100644
--- a/arch/gaudi/arm/rma_info.c
+++ b/arch/gaudi/arm/rma_info.c
@@ -12,6 +12,7 @@
 #include "uart.h"
 #include "boot_if.h"
 #include <preboot.h>
+#include <sizes.h>
 #include <hl_i2c_indirect_protocol.h>
 #include <gaudi/i2c_indirect_protocol.h>
 #include "rma_info.h"
@@ -33,7 +34,7 @@
 
 #define FLASH_STORAGE_BLOCK0	0x100000
 #define FLASH_STORAGE_BLOCK1	(FLASH_STORAGE_BLOCK0 + 0x10000)
-#define FLASH_SECTOR_SIZE	0x10000
+#define FLASH_SECTOR_SIZE	SZ_64K
 #define FLASH_NUM_FLASH_INFO	2
 #define FLASH_SECTOR_MAGIC	0xAA55AA55
 
diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index 4064a68..be93150 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -7,6 +7,7 @@
 #define COMMON_GAUDI2_ARC_GNU_H_
 
 #include <bitops.h>
+#include <sizes.h>
 
 #define PPBOOT_SIZE			0x4000		/* 16KB */
 #define PREBOOT_SIZE			0x1FC00		/* 127KB */
@@ -16,7 +17,7 @@
 
 #define HL_AXI_GENERAL_BASE	0x1000007FF0000000ULL
 #define HL_SP_SRAM_BASE		0x1000007FF7FE0000ULL
-#define HL_SP_SRAM_SIZE		0x10000
+#define HL_SP_SRAM_SIZE		SZ_64K
 
 #ifdef TARGET_IS_ONIX
 #define SDRAM_BASE		0x10000000
@@ -27,7 +28,7 @@
  * using only the first 3M of DRAM
  * located on HBM0 offset 0x0
  */
-#define ARM_SDRAM_SIZE		0x300000ull
+#define ARM_SDRAM_SIZE		(SZ_2M + SZ_1M)
 #define MEM_REGION_MASK		0x8000000000
 #else
 /*
@@ -39,7 +40,7 @@
 #define SDRAM_BASE		0x11000000
 #define SDRAM_FULL_BASE		0x1000FFFFFD000000ull
 #define ARM_ADDR_MASK		0x1FFFFFFull
-#define ARM_SDRAM_SIZE		0x3000000
+#define ARM_SDRAM_SIZE		(SZ_32M + SZ_16M)
 #define MEM_REGION_MASK		0x4000000000
 #endif
 
@@ -85,7 +86,7 @@
 #define EFUSE_SYSCFG_ENABLE_VALIDATION_MASK	BIT(10)
 
 #define RECOVERY_TEXT_BASE	(0x0 + 0x4)
-#define RECOVERY_SIZE		0x4000
+#define RECOVERY_SIZE		SZ_16K
 
 #define ARC_RST_VALUE		0x110
 #define ARC_RST_VEC_ADDR	0x0
diff --git a/arch/gaudi2/arm/common.c b/arch/gaudi2/arm/common.c
index d42e07a..92da6c3 100644
--- a/arch/gaudi2/arm/common.c
+++ b/arch/gaudi2/arm/common.c
@@ -28,6 +28,7 @@
 #include <common/pci_ids.h>
 
 #include <bitops.h>
+#include <sizes.h>
 #include <preboot.h>
 #include <pll.h>
 #include <uart.h>
@@ -46,13 +47,13 @@
 
 
 #define ARC0_ICCM_BASE_ADDR		0x1000007FF7000000ull
-#define ARC0_ICCM_SIZE			0x20000			/* 128KB */
+#define ARC0_ICCM_SIZE			SZ_128K
 #define ARC0_DCCM_BASE_ADDR		0x1000007FF7020000ull
-#define ARC0_DCCM_SIZE			0x8000			/* 32KB */
+#define ARC0_DCCM_SIZE			SZ_32K
 #define ARC1_ICCM_BASE_ADDR		0x1000007FF7040000ull
-#define ARC1_ICCM_SIZE			0x20000			/* 128KB */
+#define ARC1_ICCM_SIZE			SZ_128K
 #define ARC1_DCCM_BASE_ADDR		0x1000007FF7060000ull
-#define ARC1_DCCM_SIZE			0x8000			/* 32KB */
+#define ARC1_DCCM_SIZE			SZ_32K
 
 #define CPU1_RST_CTRL							\
 	((BIT(1) << CPU_CA53_CFG_ARM_RST_CONTROL_NCPUPORESET_SHIFT) |	\
diff --git a/arch/gaudi2/arm/common.h b/arch/gaudi2/arm/common.h
index 46f2c41..122161b 100644
--- a/arch/gaudi2/arm/common.h
+++ b/arch/gaudi2/arm/common.h
@@ -7,6 +7,7 @@
 #define COMMON_GAUDI2_ARM_H_
 
 #include <bitops.h>
+#include <sizes.h>
 
 /*
  * Gaudi2 ARM address extension is defined by bits 38,39 in the address and
@@ -47,7 +48,7 @@
 #define HL_SP_SRAM_BASE			0x3FF7FE0000ULL
 #define HL_SP_SRAM_BASE_FULL		0x1000007FF7FE0000ULL
 #define HL_SP_SRAM_INIT_CTRL		0x1000007FFCC4BD80ULL
-#define HL_SP_SRAM_SIZE			0x10000
+#define HL_SP_SRAM_SIZE			SZ_64K
 #define FUSE_WORDS_PER_BANK		0x40
 #define I2C_SLAVE_ADDRESS		0x6A
 #define I2C_SLAVE_ADDRESS_DBG		0x47
diff --git a/arch/gaudi2/common/common.h b/arch/gaudi2/common/common.h
index 04cfe2e..02f070a 100644
--- a/arch/gaudi2/common/common.h
+++ b/arch/gaudi2/common/common.h
@@ -7,6 +7,7 @@
 #define ARCH_GAUDI2_COMMON_COMMON_H_
 
 #include <preboot.h>
+#include <sizes.h>
 #include <limits.h>
 
 #ifdef TARGET_IS_PLDM
@@ -33,13 +34,13 @@
 #endif
 
 #define ARC0_ICCM_FULL_BASE			0x1000007FF7000000ull
-#define ARC0_ICCM_SIZE				0x20000			/* 128KB */
+#define ARC0_ICCM_SIZE				SZ_128K
 #define ARC0_DCCM_FULL_BASE			0x1000007FF7020000ull
-#define ARC0_DCCM_SIZE				0x8000			/* 32KB */
+#define ARC0_DCCM_SIZE				SZ_32K
 #define ARC1_ICCM_FULL_BASE			0x1000007FF7040000ull
-#define ARC1_ICCM_SIZE				0x20000			/* 128KB */
+#define ARC1_ICCM_SIZE				SZ_128K
 #define ARC1_DCCM_FULL_BASE			0x1000007FF7060000ull
-#define ARC1_DCCM_SIZE				0x8000			/* 32KB */
+#define ARC1_DCCM_SIZE				SZ_32K
 
 #define NUM_OF_DCORES			4
 #define NUM_OF_EDMA_PER_DCORE		2
diff --git a/arch/goya/arm/common.h b/arch/goya/arm/common.h
index fcf823a..308c036 100644
--- a/arch/goya/arm/common.h
+++ b/arch/goya/arm/common.h
@@ -7,6 +7,8 @@
 #ifndef COMMON_GOYA_H_
 #define COMMON_GOYA_H_
 
+#include <sizes.h>
+
 #define PPBOOT_SIZE			0x4000		/* 16KB */
 #define PREBOOT_SIZE			0x20000		/* 128KB */
 
@@ -18,7 +20,7 @@
 
 /* Scratchpad SRAM (32KB) */
 #define HL_SP_SRAM_BASE		0x7FF9000000
-#define HL_SP_SRAM_SIZE		0x8000
+#define HL_SP_SRAM_SIZE		SZ_32K
 
 #define BOOT_INFO_BASE		HL_SP_SRAM_BASE
 
diff --git a/arch/greco/arc/common.h b/arch/greco/arc/common.h
index 9b0ffdb..1e2a149 100644
--- a/arch/greco/arc/common.h
+++ b/arch/greco/arc/common.h
@@ -7,6 +7,7 @@
 #define COMMON_GRECO_ARC_GNU_H_
 
 #include <goya2_blocks_linux_driver.h>
+#include <sizes.h>
 
 #define PPBOOT_SIZE			0x4000		/* 16KB */
 #define PREBOOT_SIZE			0x10400		/* 65KB */
@@ -23,7 +24,7 @@
 #define PREBOOT_TEXT_BASE	(SDRAM_BASE + PPBOOT_SIZE + 0x8)
 
 #define HL_SP_SRAM_BASE		0x7FFBFE0000
-#define HL_SP_SRAM_SIZE		0x10000
+#define HL_SP_SRAM_SIZE		SZ_64K
 
 #define FW_WORD_NUM		0x2200	/* FW_SIZE / 2 */
 
@@ -40,7 +41,7 @@
 #define I2C_SLAVE_DEFAULT_INTR_MASK	0x48FF
 
 #define RECOVERY_TEXT_BASE	(0x0 + 0x4)
-#define RECOVERY_SIZE		0x4000
+#define RECOVERY_SIZE		SZ_16K
 
 #define HL_SPI_BASE		0x1A000000
 #define HL_CFG_BASE		0xFC000000
diff --git a/arch/greco/arc_elf32/common.h b/arch/greco/arc_elf32/common.h
index 70c0182..55e5562 100644
--- a/arch/greco/arc_elf32/common.h
+++ b/arch/greco/arc_elf32/common.h
@@ -9,6 +9,7 @@
 
 #include <goya2_blocks_linux_driver.h>
 #include <bitops.h>
+#include <sizes.h>
 
 #define PPBOOT_SIZE		0x4000		/* 16KB */
 #define PREBOOT_SIZE		0x10400		/* 65KB */
@@ -21,7 +22,7 @@
 #define PCIE_FW_SRAM_BASE	0x1BFF0000
 #define FW_SRAM_BASE		0x1B050000
 #define SP_SRAM_BASE		0x1BFE0000
-#define HL_SP_SRAM_SIZE		0x10000
+#define HL_SP_SRAM_SIZE		SZ_64K
 #define PPBOOT_TEXT_BASE	(FW_SRAM_BASE + 0xC)
 #define PREBOOT_TEXT_BASE	(SDRAM_BASE + PPBOOT_SIZE + 0x8)
 
diff --git a/arch/greco/arm/common.h b/arch/greco/arm/common.h
index 5ee44df..6c65aee 100644
--- a/arch/greco/arm/common.h
+++ b/arch/greco/arm/common.h
@@ -8,11 +8,12 @@
 
 #include <goya2_blocks_linux_driver.h>
 #include <bitops.h>
+#include <sizes.h>
 
 #define PPBOOT_SIZE			0x5000		/* 20KB */
 #define PREBOOT_SIZE			0x12800		/* 74KB */
 
-#define HL_SP_SRAM_SIZE			0x10000
+#define HL_SP_SRAM_SIZE			SZ_64K
 
 #define HL_FW_BOOT_PTRS_BASE		0x80000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xC0000
@@ -31,8 +32,8 @@
 #define HL_CFG_BASE			0x7FFC000000ull
 #define SDRAM_BASE			0x7FF0000000ull
 #define BOOT_ROM_BASE			0x7FFB040000ull
-#define BOOT_ROM_SIZE			0x10000
-#define FW_SRAM_SIZE			0x10000
+#define BOOT_ROM_SIZE			SZ_64K
+#define FW_SRAM_SIZE			SZ_64K
 
 #define FW_WORD_NUM			0x2000	/* FW_SIZE / 2 */
 
diff --git a/include/preboot.h b/include/preboot.h
index 00b7ec3..28bfb8a 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -9,6 +9,7 @@
 #define COMMON_PREBOOT_H_
 
 #include <common.h>
+#include <sizes.h>
 
 #define ARRAY_SIZE(x)		(sizeof(x) / sizeof((x)[0]))
 #define ROUND_UP4(n)		(((n) + sizeof(u32) - 1) & ~(sizeof(u32) - 1))
@@ -24,7 +25,7 @@
 #define HL_UART_BAUDRATE	115200
 #define HL_SPI_FLASH_MAX_FREQ	25000000
 
-#define FIT_IMG_SPSRAM_MAX_SIZE		0x2000
+#define FIT_IMG_SPSRAM_MAX_SIZE		SZ_8K
 #define FIT_IMG_SPSRAM_END		(HL_SP_SRAM_BASE + HL_SP_SRAM_SIZE)
 #define FIT_IMG_SPSRAM_BASE		(FIT_IMG_SPSRAM_END - \
 					 FIT_IMG_SPSRAM_MAX_SIZE)
diff --git a/include/sizes.h b/include/sizes.h
new file mode 100644
index 0000000..a1b1107
--- /dev/null
+++ b/include/sizes.h
@@ -0,0 +1,51 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+/*
+ * include/linux/sizes.h
+ */
+#ifndef __LINUX_SIZES_H__
+#define __LINUX_SIZES_H__
+
+#include <const.h>
+
+#define SZ_1				0x00000001
+#define SZ_2				0x00000002
+#define SZ_4				0x00000004
+#define SZ_8				0x00000008
+#define SZ_16				0x00000010
+#define SZ_32				0x00000020
+#define SZ_64				0x00000040
+#define SZ_128				0x00000080
+#define SZ_256				0x00000100
+#define SZ_512				0x00000200
+
+#define SZ_1K				0x00000400
+#define SZ_2K				0x00000800
+#define SZ_4K				0x00001000
+#define SZ_8K				0x00002000
+#define SZ_16K				0x00004000
+#define SZ_32K				0x00008000
+#define SZ_64K				0x00010000
+#define SZ_128K				0x00020000
+#define SZ_256K				0x00040000
+#define SZ_512K				0x00080000
+
+#define SZ_1M				0x00100000
+#define SZ_2M				0x00200000
+#define SZ_4M				0x00400000
+#define SZ_8M				0x00800000
+#define SZ_16M				0x01000000
+#define SZ_32M				0x02000000
+#define SZ_64M				0x04000000
+#define SZ_128M				0x08000000
+#define SZ_256M				0x10000000
+#define SZ_512M				0x20000000
+
+#define SZ_1G				0x40000000
+#define SZ_2G				0x80000000
+
+#define SZ_4G				_AC(0x100000000, ULL)
+#define SZ_16G				_AC(0x400000000, ULL)
+#define SZ_32G				_AC(0x800000000, ULL)
+#define SZ_64T				_AC(0x400000000000, ULL)
+
+#endif /* __LINUX_SIZES_H__ */
diff --git a/lib/common/pci.c b/lib/common/pci.c
index f65eba2..bad97cc 100644
--- a/lib/common/pci.c
+++ b/lib/common/pci.c
@@ -11,6 +11,7 @@
 #include <pcie_dbi_masks.h>
 #include <bitops.h>
 #include <preboot.h>
+#include <sizes.h>
 #include "pci.h"
 
 #define IATU_BAR_NUM_OFFSET		8
@@ -19,7 +20,7 @@
 #define IATU_MATCH_MODE_MASK		BIT(30)
 #define IATU_REGION_EN_MASK		BIT(31)
 
-#define IATU_REGION_SIZE		0x200
+#define IATU_REGION_SIZE		SZ_512
 #define IATU_REGION_INBOUND_OFF		0x100
 #define IATU_REGION_CTRL_1_OFF		0x0
 #define IATU_REGION_CTRL_2_OFF		0x4
@@ -33,7 +34,7 @@
 
 #define IATU_ACCESS_ENABLE_VAL		0x00300000
 #define IATU_ACCESS_DISABLE_VAL		0
-#define IATU_OUTBOUND_REGION_SIZE	0x00002000
+#define IATU_OUTBOUND_REGION_SIZE	SZ_8K
 #define IATU_OUTBOUND_UPPER_ADDR_VAL	0x08000000
 
 __weak u32 dev_pci_dbi_cfg_offset_get(void)
diff --git a/lib/common/spi_flash.c b/lib/common/spi_flash.c
index 7b24ed9..a55363e 100644
--- a/lib/common/spi_flash.c
+++ b/lib/common/spi_flash.c
@@ -8,6 +8,7 @@
 #include <stddef.h>
 #include <stdbool.h>
 
+#include <sizes.h>
 #include <preboot.h>
 #include "common.h"
 #include <spi.h>
@@ -30,7 +31,7 @@
 #define SPINOR_OP_ENSO			0xB1
 #define SPINOR_OP_EXSO			0xC1
 
-#define PAGE_SIZE			0x10000
+#define PAGE_SIZE			SZ_64K
 
 static struct spi_ctx flash_spi = {0};
 
diff --git a/lib/common/uart.c b/lib/common/uart.c
index 65c303b..cf17ec9 100644
--- a/lib/common/uart.c
+++ b/lib/common/uart.c
@@ -9,6 +9,7 @@
 #include <psoc_uart_0_regs.h>
 
 #include <preboot.h>
+#include <sizes.h>
 #include "uart.h"
 #include "pll.h"
 
@@ -16,7 +17,7 @@
 #define UART_SR_RXFULL			(1 << 1) /* RX Buffer Full */
 #define STR_UINT_SZ			8
 #define BITS_IN_SINGLE_CHAR		4
-#define RING_BUFF_SIZE			256
+#define RING_BUFF_SIZE			SZ_256
 
 struct ring_buf_t {
 	char buf[RING_BUFF_SIZE];
